src/components/DatePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,225 @@
+  setSelectedDate,
+  startDate,
+  endDate,
+}: CalendarProps) => {
+  const [hoveredDate, setHoveredDate] = useState<Date>();
+
+  const handleMouseOut = (): void => {
+    setHoveredDate(undefined);
+  };
+
+  return calendarBody.value.map(({ key: weekKey, value: week }) => {
+    return (
+      <tr key={weekKey}>
+        {week.map(({ date, isCurrentMonth, key: dayKey, value: fullDate }) => {
+          const isSelected =
+            (startDate && isSameDate(startDate, fullDate)) ||
+            (endDate && isSameDate(endDate, fullDate));
+
+          const today = new Date();
+
+          const isCurrentDate = isSameDate(today, fullDate);
+          const isDisabled = futureDatesDisabled ? fullDate > today : false;
+          const isBetweenStartAndEndDates = Boolean(
+            startDate && endDate && fullDate > startDate && fullDate < endDate
+          );
+
+          const shouldShowRangeIndicator =
+            !endDate &&
+            Boolean(
+              startDate && hoveredDate && fullDate > startDate && fullDate < hoveredDate
+            );
+
+          const handleMouseEnter = () => {
+            setHoveredDate(fullDate);
+          };
+
+          const handleClick = () => {
+            if (isDisabled) {
+              return false;
+            }
+            setSelectedDate(fullDate);
+
+            // User has a date range selected and clicked a new date.
+            // This will cause the selected date to be reset, thus do not close the datepicker.
+            if (startDate && endDate) {
+              return;
+            }
+
+            // The user has selected a new start date, don't close
+            if (startDate && fullDate < startDate) {
+              return;
+            }
+
+            // Only close the datepicker if the user hasn't clicked the selected start date.
+            if (startDate && !isSameDate(fullDate, startDate)) {
+              closeDatepicker();
+              return;
+            }
+          };
+          return (
+            <DateRangeTableCell
+              $shouldShowRangeIndicator={
+                shouldShowRangeIndicator || isBetweenStartAndEndDates
+              }
+              $isCurrentMonth={isCurrentMonth}
+              $isDisabled={isDisabled}
+              $isSelected={isSelected}
+              $isToday={isCurrentDate}
+              key={dayKey}
+              onClick={handleClick}
+              onMouseEnter={handleMouseEnter}
+              onMouseLeave={handleMouseOut}
+            >
+              {date}
+            </DateRangeTableCell>
+          );
+        })}
+      </tr>
+    );
+  });
+};
+
+export interface DatePickerProps {
+  endDate?: Date;
+  disabled?: boolean;
+  futureDatesDisabled?: boolean;
+  onSelectDateRange: (selectedStartDate: Date, selectedEndDate: Date) => void;
+  placeholder?: string;
+  startDate?: Date;
+}
+
+export const DateRangePicker = ({
+  endDate,
+  startDate,
+  disabled = false,
+  futureDatesDisabled = false,
```

Comment:
would be great to be able to just specify min/max dates
but that's a feature request for DatePicker, thinking out loud

Reply:
Write up a click-ui issue for this. If we find we need it in the wild, we can add it